src/main/java/fi/aalto/cs/apluscourses/model/Tutorial.java

```diff
@@ -45,4 +51,32 @@ public static Tutorial fromJsonObject(@NotNull JSONObject jsonObject) {
   public void onComplete() {
     this.tutorialCompleted.trigger();
   }
+
+  /**
+   * Downloads the module deps for a tutorial, with the module name starting with "Ideact_".
+   */
+  public abstract void downloadDependencies(@NotNull Course course, @NotNull Project project);
+
+  /**
+   * Deletes the module deps for a tutorial.
+   */
+  public abstract void deleteDependencies(@NotNull Course course, @NotNull Project project);
+
+
+  /**
+   * Returns true and shows notifications if modules are missing.
+   */
+  public abstract boolean dependenciesMissing(@NotNull Project project);
```

Comment:
Giving course or project (an IntelliJ class btw!) as an argument does not make sense here because there is already conceptual relationship between tutorials and classes/projects they belong to. Thus, they are not "free" parameters of these methods but only provided "for the sake of convenience" (giving a wrong course/project would probably lead to erroneous behavior). Thus, there almost certainly would be a better way to do it where the design would guarantee the use of correct course/project.

The use of `Project` is, as we all know, problematic anyway, but that hopefully gets fixed with #641.